composer.json

```diff
@@ -19,7 +19,9 @@
   "require-dev": {
     "laminas/laminas-cache-storage-adapter-test": "^2.4",
     "laminas/laminas-coding-standard": "~2.5.0",
+    "laminas/laminas-servicemanager": "^3.20",
```

Comment:
These are restrictions: why do we add them?

Reply:
In practice, **not** declaring these two should be better 🤔

Reply:
Without declaring either dep, we end up with `psr/container:^2`, this prevents installation of the most recent `servicemanager` which has updated types that work with psalm 5's sealed arrays. If preferable, I can just revert the dep changes and expand the baseline - in this case, as it's `require-dev`, and the deps are only relevant to a delegator factory that would only be used with SM anyhow, I figured declaring and upgrading the deps was the sane choice 🤷‍♂️

Reply:
Why is `psr/container` v2 installable here? `laminas-cache` is not compatible with v2 as it extends `PluginManager` from laminas-servicemanager and thus it is required by `laminas/laminas-cache`.
No way that `psr/container` v2 will be installed.

Reply:
There's never been a release of `servicemanager` that supports `psr/container:^2` 🤷‍♂️

However, without declaring the 2 dev deps we get:

![Screenshot 2023-01-09 at 10 50 51](https://user-images.githubusercontent.com/2803720/211291666-6d2564a5-60f2-478e-8fe1-295dbde32a3e.png)

And declaring them gives us:

![Screenshot 2023-01-09 at 10 51 14](https://user-images.githubusercontent.com/2803720/211291725-5700948c-a4b1-4563-a24b-17f1be4e2011.png)

The latter fixes the psalm issues surfaced in this PR.

It's clearly something related to symfony/console… somewhere, but like you say - `psr/container:2` should never be installable here!

Reply:
> There's never been a release of servicemanager that supports psr/container:^2 🤷‍♂️

https://github.com/laminas/laminas-servicemanager/releases/tag/3.15.0

https://github.com/laminas/laminas-servicemanager/releases/tag/3.15.1


----

laminas-cache allows Minimum of 3.11.1:
https://github.com/laminas/laminas-cache/blob/5fe4e343cf3ca5eeda1552ac6db856e2a83dbe93/composer.json#L40

Since composer is "smart", it installs 3.15.0 which allows psr/container v2 as you can also see within your Screenshot.

So the proper fix might be to raise minimum servicemanager Version in laminas-cache so that we can omit the changes here.